### ISSUE TITLE
Provide facility for machine translation of resources.json in a separate project

### ISSUE THREAD

**Issue by MrCsabaToth** [gdg-x/hoverboard#147]
Here is a simple node.js app, it uses Google Translate:
https://github.com/gdgfresno/translateResources/

I generated Spanish, Hmong and Armenian translations for ValleyDevFest 2016 to support our largest local ethnic groups.

**Comment by MrCsabaToth** [comment]
BTW, it'd be good to have the language selection popup on the top drawer too

**Comment by ozasadnyy** [comment]
@MrCsabaToth, looks interesting! How can we integrate this into Hoverboard?

**Comment by MrCsabaToth** [comment]
The best would be to make it part of the build process. Since if you add a term only into the en resource (but not the other ones) the page/site stops functioning. I didn't look at yet how I could make this part of polymer's pre-build process. Right now it's a node app, and the calls are simple translate calls (no API key needed). In my app I had Spanish, Hmong and Armenian translations.

**Comment by MrCsabaToth** [comment]
Thinking more: this machine translation probably should not overwrite non machine translations, so probably we need to protect those entries in the resource file somehow. So we need to think more about that.

**Comment by ozasadnyy** [comment]
Seems like time to use [custom build](https://github.com/PolymerElements/generator-polymer-init-custom-build) for Polymer CLI

**Comment by MrCsabaToth** [comment]
Since `resources.json` is not a flat key-value pair list any more, I need to enhance the translation script first.

**Comment by ozasadnyy** [comment]
@MrCsabaToth can you make script translate only string values?

**Comment by abraham** [comment]
This sounds neat but it's implementation will be very dependent on #454. Seeing as this is so old I'm going to close it. Once #454 is completed if there is still interest a new issue can be created.
